src/Uno.UI/Extensions/PermissionsHelper.Android.cs

```diff
@@ -19,11 +19,185 @@ public static void Initialize()
 					.CreateAsync<string, bool>(TryGetPermissionCore)
 					.LockInvocation(InvocationLockingMode.Share);
 
-			Windows.Extensions.PermissionsHelper.Initialize(getPermission, CheckPermission);
-		}
+			Windows.Extensions.PermissionsHelper.Initialize(getPermission, CheckPermission, MissingPermissions, TryAskPermissionAsync);
+		}
+
+		private static int _permissionRequest;
+		#region "submethods for MissingPermissions"
+		private static IList<string> GetManifestPermissions()
+		{
+			// get all permission declared in Manifest (can be null)
+			Android.Content.Context context = Android.App.Application.Context;
+			Android.Content.PM.PackageInfo packageInfo =
+				context.PackageManager.GetPackageInfo(context.PackageName, Android.Content.PM.PackageInfoFlags.Permissions);
+			return packageInfo?.RequestedPermissions;
+		}
+
+		private static bool AreAllPermissionDeclared(IList<string> manifestPermissions, string[] requiredPermissions)
+		{
+			// check if all requiredPermissions are declared in Manifest 
+			if (requiredPermissions is null)
+			{
+				return true;    // no required permissions, so - everything is OK
+			}
+
+			foreach (string permission in requiredPermissions)
+			{
+				if (!manifestPermissions.Any(p => p.Equals(permission, StringComparison.OrdinalIgnoreCase)))
+				{
+					return false;
+				}
+			}
+
+			return true;
+
+		}
+
+		private static List<string> PermissionWeWant(IList<string> manifestPermissions, string[] requiredPermissions, string[] optionalPermissions)

```

Comment:
```suggestion
		private static List<string> GetRequestedPermissions(IList<string> manifestPermissions, string[] requiredPermissions, string[] optionalPermissions)

```

Reply:
"Requested" is misleading - requested by whom from whom, and when?
Second parameter is list of permissions that "maybe", so not requested.
And returned list is list of permission we want to ask for, so "ToBeRequested" :)

Reply:
That's fair. Lets call it `GetPermissionsToRequest` then.

Reply:
It is not permissions to request, because some of them are already granted and would be filtered-out from this list.

Reply:
If the list returned the items to be requested, it's a list of items to request.

Reply:
No, we doesn't request this list of permissions - before asking for these permissions, this list is filtered.